sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -174,4 +174,36 @@ class CollationSuite extends QueryTest with SharedSparkSession {
           Row(expected))
     }
   }
+
+
+  test("disable partition on collated string column") {
+    def createTable(partitionColumns: String*): Unit = {
+      val tableName = "test_partition_tbl"
+      withTable(tableName) {
+        sql(
+          s"""
+             |CREATE TABLE $tableName
```

Comment:
can we also test v2 tables? See `DataSourceV2SQLSuite` about how we create and test v2 tables using v2 catalogs.

Reply:
it seems that we don't even check if partitioning is allowed or not for v2 table creation ([code](https://github.com/apache/spark/blob/a939a7d0fd9c6b23c879cbee05275c6fbc939e38/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala#L251)), is this a bug?

Reply:
nvm, v2 table does not use directory names to implement partitioning. We don't need to test it.